### ISSUE TITLE
Questions: How can I ensure that my workflow works perfectly?

### ISSUE THREAD

**Issue by howard-yeh** [tinkerbell/charts#64]
Hi there, I would like to inquire about Tinkerbell provisioning questions, and here are the questions:
1.  Should Tinkerbell have to install tink-cli to manage workflow, hardware, and monitor its events?
2.  When I use ipxe.yml to reboot the target Arm server and start to provision, the target can enter the grub menu, but it cannot load the target OS (e.g., Ubuntu 20.04). I also checked the logs in each component of Tinkerbell, but they do not show any errors.  I have already collected the logs from tink-controller and tink-server, and they are shown below. By the way, I have installed vmlinuz-aarch64, initramfs-aarch64, initramfs-x86_64, and vmlinuz-x86_64. Also, the Ubuntu image has already been converted into raw.gz (focal-server-cloudimg-arm64.raw.gz/focal-server-cloudimg-amd64.raw.gz).
3. When I provision an x86 machine, we encountered PXE-99: Unexpected network error, and here are the full logs from the target server.

## Expected Behaviour

The "Ubuntu 20.04" option is not appearing in the grub menu, and there are no errors shown in the log.

## Current Behaviour

The grub menu did not show the "Ubuntu 20.04" option, and there were no errors shown in the log.

## Steps to Reproduce (for bugs)

1. deploy tinkerbell pods by helm stack
2. apply hardware.yml
3. apply template.yml
4. apply workflow.yml
5. apply ipxe.yml (which control BMC and set the ipxe booting mode)

Note: The workflow state always shows as "Pending."

## Your Environment

* Operating System and version (e.g. Linux, Windows, MacOS):
Operating System: Linux (Ubuntu 20.04)

* How are you running Tinkerbell? Using Vagrant & VirtualBox, Vagrant & Libvirt, on Packet using Terraform, or give details:
Deploy by kubernets (helm) approach

tink-controller log:
|--------|
| {"level":"info","ts":1695707823.051005,"logger":"github.com/tinkerbell/tink","caller":"tink-controller/main.go:120","msg":"no config file found"}
{"level":"info","ts":1695707823.05112,"logger":"github.com/tinkerbell/tink","caller":"tink-controller/main.go:70","msg":"Starting controller version devel"}
{"level":"info","ts":1695707823.0534592,"logger":"github.com/tinkerbell/tink","caller":"manager/internal.go:360","msg":"Starting server","kind":"health probe","addr":"[::]:8081"}
{"level":"info","ts":1695707823.0535295,"logger":"github.com/tinkerbell/tink","caller":"manager/server.go:50","msg":"starting server","path":"/metrics","kind":"metrics","addr":"[::]:8080"}
{"level":"info","ts":1695707823.0536685,"logger":"github.com/tinkerbell/tink","caller":"controller/controller.go:177","msg":"Starting EventSource","controller":"workflow","controllerGroup":"tinkerbell.org","controllerKind":"Workflow","source":"kind source: *v1alpha1.Workflow"}
{"level":"info","ts":1695707823.0537043,"logger":"github.com/tinkerbell/tink","caller":"controller/controller.go:185","msg":"Starting Controller","controller":"workflow","controllerGroup":"tinkerbell.org","controllerKind":"Workflow"}
{"level":"info","ts":1695707823.1586273,"logger":"github.com/tinkerbell/tink","caller":"controller/controller.go:219","msg":"Starting workers","controller":"workflow","controllerGroup":"tinkerbell.org","controllerKind":"Workflow","worker count":1}
{"level":"info","ts":1695707823.1588185,"logger":"github.com/tinkerbell/tink","caller":"workflow/reconciler.go:38","msg":"Reconciling","controller":"workflow","controllerGroup":"tinkerbell.org","controllerKind":"Workflow","Workflow":{"name":"smc-arm-workflow","namespace":"tink-system"},"namespace":"tink-system","name":"smc-arm-workflow","reconcileID":"acc48843-fff4-436f-b5d4-1239aea0a2d6"}| 

tink-server log:
|--------|
| {"level":"info","ts":1695707823.8072968,"logger":"github.com/tinkerbell/tink","caller":"tink-server/main.go:178","msg":"No config file found"}
{"level":"info","ts":1695707823.8073664,"logger":"github.com/tinkerbell/tink","caller":"tink-server/main.go:96","msg":"Starting version devel"}
{"level":"info","ts":1695707823.8500073,"logger":"github.com/tinkerbell/tink","caller":"tink-server/main.go:137","msg":"started listener","address":"[::]:42113"}
{"level":"info","ts":1695707823.8501341,"logger":"github.com/tinkerbell/tink","caller":"httpserver/http_server.go:31","msg":"serving http"} |

**Comment by jacobweinstock** [comment]
Hey @howard-yeh, apologies for the delayed response.

_1. Should Tinkerbell have to install tink-cli to manage workflow, hardware, and monitor its events?_
There is not tink-cli. You can use kubectl to manage hardware, templates, and workflows.  Workflow events can be seen on the workflow object and via Boots/Smee and Tink controller.

Would you mind sharing the Boots/Smee logs? and any screen shots from the machines console?

**Comment by howard-yeh** [comment]
Hi @jacobweinstock, thanks for reply this issue.

After reviewing the code and logs, I discovered that my tinkerbell-client was unable to pull down the tink-worker image via Docker. The root cause was due to being in an air-gap environment, where connections to the public network are not allowed. As a workaround, I manually modified the /etc/hosts file using 'docker-shell'.

The related issue reference: https://github.com/tinkerbell/sandbox/issues/178
